### PR TITLE
Plugins: Yan LR & YAML Smith

### PR DESCRIPTION
**Pull request by sanssecours**
# Checklist

- [x] I added unit tests.
- [x] I ran all tests locally and everything went fine.
- [x] The new code contains code comments and logging statements.

### CODE REVIEW

File: src/plugins/yanlr/README.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
+- infos = Information about the yanlr plugin is in keys below
+- infos/author = René Schwaiger <sanssecours@me.com>
+- infos/licence = BSD
+- infos/needs = directoryvalue yamlsmith
```
Comment (markus2330) [inline src/plugins/yanlr/README.md]: Does this work? (To use two different storage plugins at the same time)

Why the split? (Do you want to share yamlsmith across all parsers?)

File: src/plugins/yanlr/README.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+- an ANTLR [parser grammar](YAML.g4),
+- a [listener](listener.cpp), that reacts to matches in the grammar building a key set, and
+- an [error listener](error_listener.cpp) that reacts to syntax errors
+
+.
+
+## Comments
+
+The [lexer](yaml_lexer.cpp) does currently tokenize comments. Consequently the [plugin grammar](YAML.g4) of the plugin does also match comments. However, the [listener](listener.cpp) does currently **ignore comments**.
```
Comment (markus2330) [inline src/plugins/yanlr/README.md]: Move to limitations?

File: src/plugins/yanlr/README.md (hunk 1 above)
Comment (sanssecours) [reply src/plugins/yanlr/README.md]: > Does this work?

Yes.

> Do you want to share yamlsmith across all parsers?

Yep, YAML Smith should act as serialization plugin for Yan LR and other upcoming YAML plugins.

**Comment by markus2330** [comment]
I removed the ready to merge because the build servers are still building. Please readd.

**Pull request merged**